### ISSUE TITLE
Sublime Text Syntax Highlighting

### ISSUE THREAD

**Issue by waterstone15** [DanielXMoore/Civet#422]
I'd like to start migrating some CoffeeScript projects to civet. I've tried to create a sublime text syntax file, but that's pretty far outside my wheelhouse. After a failing at that for few hours I've landed here seeking help.

Any chance a sublime text syntax highlighting package is on the docket?

**Comment by edemaine** [comment]
According to https://www.sublimetext.com/docs/syntax.html, Sublime supports TextMate grammars.  [Civet's TextMate grammar](https://github.com/DanielXMoore/Civet/tree/main/lsp/syntaxes) is far from good, but it's better than nothing. Can you see whether it's possible to install it as is? Perhaps via Tools -> Developer -> New Syntax.

**Comment by waterstone15** [comment]
@edmaine Yes, in theory – I was trying to use `civet.json` (I tried repackaging it as .sublime-package and .sublime-syntax, converting it .yaml and .xml, etc.) but after thorough floundering I came here.

_ps. 'Tools → Developer → New Syntax' creates a .sublime-syntax .yaml file, which afaik is not a compatible format with `civet.json`s format._

**Comment by edemaine** [comment]
It might make sense to ask the Sublime community for suggestions on how to import a JSON TextMate grammar. I assume this is a general (and hopefully solved) problem. Please report back if you find out anything!

**Comment by waterstone15** [comment]
In theory ST supports XML/plist `.tmLanguage` files, but when I try to convert the `civet.json` to XML the output doesn't seem to be something ST can read.

**Comment by Kristinita** [comment]
**Type: Opinion**

Possibly, the lack of the IDE plugins is the main reason why Civet hasn’t popularity it deserves.

According to the official site, Civet has solely [**the extension for VS Code**](https://civet.dev/integrations). I don’t want to install VS Code especially for Civet, and I think that many users have the similar behavior. I don’t think that limiting the Civet community to the VS Code users is the best idea for both users and the project.

I would give a high priority to the task of creating (or hiring people who can do it) plugins for the most popular IDEs. At least the syntax highlighting is required for users, and having the [**LSP implementation**](https://lsp.sublimetext.io/language_servers/) would be helpful.

Thanks.
